### PR TITLE
Work/craft 3831 starflow

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Review by bepri — CHANGES_REQUESTED** [review]
Nice! In my testing, I was able to use the new makefile to create a dev environment from scratch, run the linters, and then package rockcraft's snap. Just a couple potential stragglers I spotted.

File: common.mk [hunk 1]
```diff
@@ -46,7 +46,7 @@ setup: install-uv setup-precommit ## Set up a development environment
 
 .PHONY: setup-tests
 setup-tests: install-uv install-build-deps ##- Set up a testing environment without linters
-	uv sync --frozen
+	uv sync --frozen $(SETUP_TESTS_EXTRA_ARGS)
```
[the remark this answers is not in the document]
Comment (tigarmo) [reply common.mk:49]: yes, on the [Makefile](https://github.com/canonical/rockcraft/pull/782/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R9).

[the remark this answers is not in the document]
Comment (lengau) [reply common.mk:49]: Let's upstream this :-)

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -100,25 +100,28 @@ parts:
...
+        - "CFLAGS": "$(pkg-config python-3.12 yaml-0.1 --cflags)"
         - "PIP_NO_BINARY": ":all:"
+        - UV_FROZEN: "true"
+        - UV_PYTHON: /usr/bin/python3
     organize:
         bin/craftctl: libexec/rockcraft/craftctl
         # Extensions currently expect to find data files in share/rockcraft/
         "**/site-packages/extensions": share/rockcraft/extensions
     override-build: |
+      uv export --no-dev --no-emit-workspace --no-emit-package pywin32 --output-file uv-requirements.txt
```
Comment (lengau) [inline snap/snapcraft.yaml:124]: This pywin32 item is interesting. Do you know where it came from?

**Review by lengau — APPROVED** [review]
LGTM, thanks! I agree that the pylint changes should be in a separate PR

File: snap/snapcraft.yaml (hunk 2 above)
Comment (tigarmo) [reply snap/snapcraft.yaml:124]: kind of; craft-cli [depends on pywin32](https://github.com/canonical/craft-cli/blob/main/pyproject.toml#L11) when in win32, and somehow `uv export` is adding that dependency even on `!win32`

**Review by bepri — APPROVED** [review]

**Pull request merged**